src/components/visualizations/chart/highcharts/plugins/adjustTickAmount.ts

```diff
@@ -0,0 +1,245 @@
+    return axis.coll === "yAxis";
+}
+
+/**
+ * Check if user sets min/max on any axis
+ * @param chart
+ */
+function isUserSetExtremesOnAxes(chart: Highcharts.Chart): boolean {
+    const yAxes = chart.userOptions.yAxis;
+    return yAxes[0].isUserMinMax || yAxes[1].isUserMinMax;
+}
+
+/**
+ * Get direction to make secondary axis align to primary axis
+ * @param primaryAxis
+ * @param secondaryAxis
+ * @return
+ *     -1: move zero index to left
+ *      0: it aligns
+ *      1: move zero index to right
+ */
+function getDirection(primaryAxis: IHighchartsAxisExtend, secondaryAxis: IHighchartsAxisExtend): number {
+    const { tickPositions: primaryTickPosition } = primaryAxis;
+    const { tickPositions: secondaryTickPosition } = secondaryAxis;
+
+    const primaryZeroIndex = primaryTickPosition.indexOf(0);
+    const secondaryZeroIndex = secondaryTickPosition.indexOf(0);
+
+    // no need to align zero on axes without zero or aligned already
+    if (
+        isNil(primaryZeroIndex) ||
+        isNil(secondaryZeroIndex) ||
+        primaryZeroIndex < 0 ||
+        secondaryZeroIndex < 0 ||
+        primaryZeroIndex === secondaryZeroIndex
+    ) {
+        return ALIGNED;
+    }
+
+    if (primaryZeroIndex > secondaryZeroIndex) {
+        return MOVE_ZERO_RIGHT;
+    }
+
+    return MOVE_ZERO_LEFT;
+}
+
+/**
+ * Add new tick to first or last position
+ * @param tickPositions
+ * @param tickInterval
+ * @param isAddFirst: if true, add to first. Otherwise, add to last
+ */
+function addTick(tickPositions: number[], tickInterval: number, isAddFirst: boolean): number[] {
+    const tick: number = isAddFirst
+        ? correctFloat(tickPositions[0] - tickInterval)
+        : correctFloat(tickPositions[tickPositions.length - 1] + tickInterval);
+
+    return isAddFirst ? [tick, ...tickPositions] : [...tickPositions, tick];
+}
+
+/**
+ * Add or reduce ticks
+ * @param axis
+ */
+export function adjustTicks(axis: IHighchartsAxisExtend): void {
+    let tickPositions: number[] = (axis.tickPositions || []).slice();
+    const tickAmount: number = axis.tickAmount;
+    const currentTickAmount: number = tickPositions.length;
+
+    if (currentTickAmount === tickAmount) {
+        return;
+    }
+
+    // add ticks to either start or end
+    if (currentTickAmount < tickAmount) {
+        const min = axis.min;
+        const tickInterval = axis.tickInterval;
+
+        while (tickPositions.length < tickAmount) {
+            if (axis.dataMax <= 0 || axis.max <= 0) {
+                tickPositions = addTick(tickPositions, tickInterval, true);
+            } else if (axis.dataMin >= 0 || axis.min >= 0) {
+                tickPositions = addTick(tickPositions, tickInterval, false);
+            } else if (tickPositions.length % 2 || min === 0) {
+                // default HC behavior
+                tickPositions = addTick(tickPositions, tickInterval, false);
+            } else {
+                tickPositions = addTick(tickPositions, tickInterval, true);
```

Comment:
Sorry for missing it.
I  think it can be simplify as following:
```
while (tickPositions.length < tickAmount) {
     const isAddFirst = !(axis.dataMin >= 0 || axis.min >= 0 || min === 0 || tickPositions.length % 2);
     tickPositions = addTick(tickPositions, tickInterval, isAddFirst);
}
```